### ISSUE TITLE
Getting spacy/model info as dictionary

### ISSUE THREAD

**Issue by chozelinek** [explosion/spaCy#2296]
## Feature description

It could be helpful to get the information about spaCy or the models as a dictionary when using `spacy.info()`, a use case would be to record programmatically information about spaCy and/or a model. Similarly as we can do with the `meta` attribute of a `Language` object.

My proposal is below. If you think it is useful I can make a PR.

In `spacy/__init__.py`

```python
def info(model=None, markdown=False, print=True):
    return cli_info(model, markdown, print)
```

In `spacy/cli/info.py`

```python
def info(model=None, markdown=False, print=True):
    """Print info about spaCy installation. If a model shortcut link is
    speficied as an argument, print model information. Flag --markdown
    prints details in Markdown for easy copy-pasting to GitHub issues.
    """
    if model:
        if util.is_package(model):
            model_path = util.get_package_path(model)
        else:
            model_path = util.get_data_path() / model
        meta_path = model_path / 'meta.json'
        if not meta_path.is_file():
            util.prints(meta_path, title=Messages.M020, exits=1)
        meta = util.read_json(meta_path)
        if model_path.resolve() != model_path:
            meta['link'] = path2str(model_path)
            meta['source'] = path2str(model_path.resolve())
        else:
            meta['source'] = path2str(model_path)
        if print:
            print_info(meta, 'model %s' % model, markdown)
        else:
            return meta
    else:
        data = {'spaCy version': about.__version__,
                'Location': path2str(Path(__file__).parent.parent),
                'Platform': platform.platform(),
                'Python version': platform.python_version(),
                'Models': list_models()}
        if print:
            print_info(data, 'spaCy', markdown)
        else:
            return data
```

**Comment by ines** [comment]
Thanks a lot for the very detailed feature request!

And good timing – something similar was already requested in #2196 I actually just implemented the feature a few days ago (see 3c80f69ff51d4dbdb3f1a7849fd8fe6f40853cb4), so it will be shipped with the next spaCy release 🎉 I think my implementation is almost identical to yours, too – I only reversed the conditional and added a `--silent` flag instead (because as far as I know, flags in the Plac CLI can't default to `True`).

**Comment by chozelinek** [comment]
Glad to hear that the feature will be there soon! Looking forward to it!
